docs/virtual_machines/disks_and_volumes.md

```diff
@@ -1468,3 +1468,136 @@ This can be done in using startup script. See [cloudInitNoCloud](#cloudInitNoClo
+> to use hostpaths. This feature is available starting with KubeVirt v0.36.0.
+
+We need a `StorageClass` which uses the provider `no-provisioner`:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: no-provisioner-storage-class
+provisioner: kubernetes.io/no-provisioner
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer
+```
+
+To make the shared directory available for VMs, we need to create a PV and a PVC that could be consumed by the VMs:
+
+```yaml
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: hostpath
+spec:
+  capacity:
+    storage: 10Gi
+  accessModes:
+    - ReadWriteMany
+  hostPath:
+    path: "/tmp/data"
+  storageClassName: "no-provisioner-storage-class"
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - node01
+---  
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: hostpath-claim
+spec:
+  accessModes:
+    - ReadWriteMany
+  storageClassName: "no-provisioner-storage-class"
+  resources:
+    requests:
+      storage: 10Gi
+```
+> **Note:** Change the `node01` value for the node name where you want the shared directory will be located.
+
+The VM definition have to request the PVC `hostpath-claim` and attaching it as a virtiofs filesystem:
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: hostpath-vm
+  name: hostpath
+spec:
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt.io/domain: hostpath
+        kubevirt.io/vm: hostpath
+    spec:
+      domain:
+        cpu:
+          cores: 1
+          sockets: 1
+          threads: 1
+        devices:
+          filesystems:
+            - name: vm-hostpath
+              virtiofs: {}
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+            - name: default
+              masquerade: {}
+          rng: {}
+        resources:
+          requests:
+            memory: 1Gi
+      networks:
+        - name: default
+          pod: {}
+      terminationGracePeriodSeconds: 180
+      volumes:
+        - containerDisk:
+            image: quay.io/containerdisks/fedora:latest
+          name: containerdisk
+        - cloudInitNoCloud:
+            userData: |-
+              #cloud-config
+              chpasswd:
+                expire: false
+              password: password
+              user: fedora
+          name: cloudinitdisk
+        - name: vm-hostpath
+          persistentVolumeClaim:
+            claimName: hostpath-claim
+```
+
+
+#### Hostpath Mounting
+To mount the hostpath in the VM, we have to access the node for setting the proper SELinux context to the shared directory:
```

Comment:
I can't follow why we need to do this after we create the PV and PVC? Wouldn't this be something we would need to prepare before?

Reply:
The system does not set the SELinux context by itself and as far as I know, this is an intended behavior. @xpivarc Can you provide further details on this please?

Reply:
Could be possible to start with setting up the directory labels?, I mean, something like (not intended to be copied verbatim :) just a rough idea):
"
Let's assume we want to share the node directory `/mnt/share`, first we must check the security label of  `/mnt/share`, otherwise the VM will not be able to access that directory:
``` example of to check sec label ``
if the sec label isn't present or is not `container_file_t` we need to label it as:
``` sudo chcon ... ```
"

So, I think a structure like this is a bit clearer:
- Checking/Setting up the shared directory
- all the yamls
- Hostpath mounting (with instructions from inside the VM only)

Reply:
I need to admit I have never tried with hostpath, but with a proper storage class, the selinux label should be handled directly from the CSI plugin. Don't we want to illustrate a more generic example with PVCs?

Reply:
> The system does not set the SELinux context by itself and as far as I know, this is an intended behavior. @xpivarc Can you provide further details on this please?

Yes, but at the same time, it depends on what approach you choose. See the above comments.

Reply:
@xpivarc yes, I agree with you.
I understand that this initial example is targeting hostpath specifically. But on the other hand, we don't have any other good examples with virtiofs, and I think it is important to explain to the users that they have other (better) solutions. 
So, as @xpivarc already said, I think it is important to mention the other cases

Reply:
Thanks for your comments. Regarding @germag suggestion, I agree and I have changed the structure as described. I think this structural change could solve some of your concerns.
About the generic example with PVCs, do you mean an example like [this](https://github.com/kubevirt/user-guide/blob/main/docs/virtual_machines/disks_and_volumes.md?plain=1#L1406-L1470)? I have reworked a bit the existing [example](https://github.com/kubevirt/user-guide/blob/main/docs/virtual_machines/disks_and_volumes.md?plain=1#L1406-L1470) and mine to reflect your suggestions. Furthermore, I have added a few lines explaining that the hostpath method is discouraged.
Additionally, I guess that the third method mentioned is using a proper storage class. However, I have never tried this approach and I'm not sure how to do it. If you have some hints in this regard please, let me know. On the other hand, because I need some time to learn and test it, I will post it in a separate PR.